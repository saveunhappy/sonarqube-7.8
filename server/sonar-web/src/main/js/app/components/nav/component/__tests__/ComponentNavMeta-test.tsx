/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import { ComponentNavMeta, getCurrentPage, Props } from '../ComponentNavMeta';
import {
  mockShortLivingBranch,
  mockComponent,
  mockLongLivingBranch,
  mockPullRequest,
  mockCurrentUser,
  mockLoggedInUser
} from '../../../../../helpers/testMocks';

describe('#ComponentNavMeta', () => {
  it('renders status of short-living branch', () => {
    expect(shallowRender()).toMatchSnapshot();
  });

  it('renders meta for long-living branch', () => {
    expect(
      shallowRender({ branchLike: mockLongLivingBranch(), currentUser: mockLoggedInUser() })
    ).toMatchSnapshot();
  });

  it('renders meta for pull request', () => {
    expect(
      shallowRender({
        branchLike: mockPullRequest({
          url: 'https://example.com/pull/1234'
        })
      })
    ).toMatchSnapshot();
  });
});

describe('#getCurrentPage', () => {
  it('should return a portfolio page', () => {
    expect(getCurrentPage(mockComponent({ key: 'foo', qualifier: 'VW' }), undefined)).toEqual({
      type: 'PORTFOLIO',
      component: 'foo'
    });
  });

  it('should return an app page', () => {
    expect(
      getCurrentPage(
        mockComponent({ key: 'foo', qualifier: 'APP' }),
        mockLongLivingBranch({ name: 'develop' })
      )
    ).toEqual({ type: 'APPLICATION', component: 'foo', branch: 'develop' });
  });

  it('should return a portfolio page', () => {
    expect(getCurrentPage(mockComponent(), mockShortLivingBranch())).toEqual({
      type: 'PROJECT',
      component: 'my-project',
      branch: undefined
    });
  });
});

function shallowRender(props: Partial<Props> = {}) {
  return shallow(
    <ComponentNavMeta
      branchLike={mockShortLivingBranch()}
      component={mockComponent({ analysisDate: '2017-01-02T00:00:00.000Z', version: '0.0.1' })}
      currentUser={mockCurrentUser()}
      warnings={[]}
      {...props}
    />
  );
}
