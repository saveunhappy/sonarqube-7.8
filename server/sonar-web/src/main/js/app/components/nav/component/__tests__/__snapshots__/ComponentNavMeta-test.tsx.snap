// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ComponentNavMeta renders meta for long-living branch 1`] = `
<div
  className="navbar-context-meta flex-0"
>
  <div
    className="spacer-left text-ellipsis"
  >
    <DateTimeFormatter
      date="2017-01-02T00:00:00.000Z"
    />
  </div>
  <Tooltip
    mouseEnterDelay={0.5}
    overlay="version 0.0.1"
  >
    <div
      className="spacer-left text-limited"
    >
      version
       
      0.0.1
    </div>
  </Tooltip>
  <div
    className="navbar-context-meta-secondary"
  >
    <Connect(HomePageSelect)
      className="spacer-left"
      currentPage={
        Object {
          "branch": "branch-6.7",
          "component": "my-project",
          "type": "PROJECT",
        }
      }
    />
  </div>
</div>
`;

exports[`#ComponentNavMeta renders meta for pull request 1`] = `
<div
  className="navbar-context-meta flex-0"
>
  <div
    className="spacer-left text-ellipsis"
  >
    <DateTimeFormatter
      date="2017-01-02T00:00:00.000Z"
    />
  </div>
  <div
    className="navbar-context-meta-secondary display-inline-flex-center"
  >
    <a
      className="display-inline-flex-center big-spacer-right"
      href="https://example.com/pull/1234"
      rel="noopener noreferrer"
      target="_blank"
    >
      branches.see_the_pr
      <DetachIcon
        className="little-spacer-left"
        size={12}
      />
    </a>
    <Connect(BranchStatus)
      branchLike={
        Object {
          "analysisDate": "2018-01-01",
          "base": "master",
          "branch": "feature/foo/bar",
          "key": "1001",
          "target": "master",
          "title": "Foo Bar feature",
          "url": "https://example.com/pull/1234",
        }
      }
      component="my-project"
    />
  </div>
</div>
`;

exports[`#ComponentNavMeta renders status of short-living branch 1`] = `
<div
  className="navbar-context-meta flex-0"
>
  <div
    className="spacer-left text-ellipsis"
  >
    <DateTimeFormatter
      date="2017-01-02T00:00:00.000Z"
    />
  </div>
  <div
    className="navbar-context-meta-secondary display-inline-flex-center"
  >
    <Connect(BranchStatus)
      branchLike={
        Object {
          "analysisDate": "2018-01-01",
          "isMain": false,
          "mergeBranch": "master",
          "name": "feature/foo",
          "type": "SHORT",
        }
      }
      component="my-project"
    />
  </div>
</div>
`;
